### DISCUSSION TITLE
post/powerautomate-update-locked-files/

### DISCUSSION THREAD

**Discussion by giscus** [AlexanderHenkel/blogComments#9] [Announcements]
# post/powerautomate-update-locked-files/

Learn how to update columns for files in libraries that are locked/checked out/need approval without raising the version number

https://alexanderhenkel.dk/post/powerautomate-update-locked-files/

**Comment by blumenhause** [comment]
I tried this method instead of using Update file properties action but it doesn't seem to work for me. The only time it works is if the user initiating flow is same as the user who checked out the file, but in my case flow would be run by a service account. This service account has highest privilege in the SPO site and library but still fails. The error on the Send HTTP Request to SharePoint action is "Bad Request" and it keeps retrying.

**Reply by AlexanderHenkel** [reply]
I am afraid you are correct, that the metadata can not be updated if another user has checked out the file.
I tried to insert in the **editor** field the following expression:
```if(outputs('Get_file_properties')['body/{IsCheckedOut}'], outputs('Get_file_properties')['body/CheckoutUser']['Email'], 'admin@xxx.onmicrosoft.com')```
Hoping that if the editor was the same as the user who checked out the file was the same, it would work, but I also got a bad request.
I am afraid that with Power Automate we are forced to use the **Discard checkout** action, or use the **Check-in** action, with the *overwrite* option

**Comment by SunnyTLL** [comment]
Great article, thank you this helped me alot!

**Reply by AlexanderHenkel** [reply]
So happy to hear that. I use this method weekly, instead of the standard action

**Comment by orionn2o** [comment]
This seemed to fail for me. The field updated correctly, but it ultimately seemed to kick the user out of the document and flag that the documents could no longer sync meaning that they would lose their changes. Any ideas?

**Reply by orionn2o** [reply]
Note after a bit of playing, updating this :

    "bNewDocumentUpdate": true

to 

    "bNewDocumentUpdate": false

seemed to fix my issue. I'm not 100% clear on what that does or what the impact is yet

**Reply by AlexanderHenkel** [reply]
```"bNewDocumentUpdate": true```

Updates the version history of the file.

Looking forward to hear your further experience

**Comment by frizz3-cloud** [comment]
Hi, I have an issue with using this method. It works perfectly if I run my flow, but if I share it for others to use it fails at updating approval status with 403 status, access denied. Any clue why and how to work around it?

**Reply by AlexanderHenkel** [reply]
Did you setup the editor to be your service account, and did you change the “run only user” to that account?
